### PR TITLE
Create storage container on workspace creation (WOR-92).

### PR DESCRIPTION
**Pull request by cahrens**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala [hunk 1]
```diff
@@ -13,6 +13,9 @@ import java.util.UUID
 import scala.collection.concurrent.TrieMap
 import scala.jdk.CollectionConverters._
 
+object MockConstants {
```
Comment (cahrens) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala]: Let me know if there's a better way to do this in Scala.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala [hunk 2]
```diff
@@ -13,6 +13,9 @@ import java.util.UUID
...
+  final def storageAccountId: UUID = UUID.fromString("11111111-2222-3333-5555-666666666666")
```
Comment (aherbst-broad) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala]: You can make this a val and forego the final qualifier:

```suggestion
  val storageAccountId: UUID = UUID.fromString("11111111-2222-3333-5555-666666666666")
```

This raises the question though, why does this need to be a constant?

**Review by aherbst-broad — APPROVED** [review]

File: core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala (hunk 2 above)
Comment (cahrens) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala]: The issue is that the account ID passed back from creating the storage account is what will be passed to the creation request for the container. If I don't make it a constant, I don't have a way to verify that the right thing was passed into the container creation.

See https://github.com/broadinstitute/rawls/pull/1857/files#diff-b8196c4af84188faa6eb1f65595077eb0b25f6c06f5a8e64b5c60e4832b93118R198

Comment (marctalbott) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala]: You could override the `MockWorkspaceManagerDAO` to return a specified storage account id. Something like this should work:

```
val storageAccountId = UUID.randomUUID()
val customWsmDao = new MockWorkspaceManagerDAO() {
  override def mockCreateAzureStorageAccountResult() = new CreatedControlledAzureStorage().
    resourceId(storageAccountId).azureStorage(new AzureStorageResource()) 
}
val workspaceManagerDAO = Mockito.spy(customWsmDao)
```

Comment (marctalbott) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala]: Another alternative would be to pass it as a parameter to `MockWorkspaceManagerDAO` and have it default to a random UUID

**Review by marctalbott — APPROVED** [review]
Looks good! Couple of untested suggestions for dealing with the constant test data, but just switching it to a val would be fine as well if that's preferred

File: core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala (hunk 2 above)
Comment (cahrens) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/mock/MockWorkspaceManagerDAO.scala]: Thanks @marctalbott, this is a much nicer approach.

**Pull request merged**